core/domain/suggestion_jobs_one_off.py

```diff
@@ -72,3 +77,138 @@ def reduce(key, value):
         if key == 'SUCCESS':
             value = len(value)
         yield (key, value)
+
+
+class PopulateTranslationContributionStatsOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """A reusable one-time job that may be used to populate translation
+    suggestion related counts in the TranslationContributionStatsModel.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function (generator). Computes word counts of
+        translations suggestions and outputs suggestion metadata.
+
+        Args:
+            item: GeneralSuggestionModel. An instance of GeneralSuggestionModel.
+
+        Yields:
+            tuple(key, recent_activity_commits). Where:
+                key: str. The entity ID of the corresponding
+                    TranslationContributionStatsModel.
+                dict. Has the keys:
+                    suggestion_status: str. The translation suggestion status.
+                    edited_by_reviewer: bool. Whether the translation suggestion
+                        was edited by a reviewer.
+                    content_word_count: int. The word count of the translation
+                        suggestion content HTML.
+                    last_updated_date: date. The last updated date of the
+                        translation suggestion.
+        """
+        if item.suggestion_type != feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT:
+            return
+
+        suggestion = suggestion_services.get_suggestion_from_model(item)
+
+        # Try to extract the topic ID from the corresponding exploration
+        # opportunity.
+        topic_id = ''
```

Comment:
@sagangwee can I know why we are setting empty string for topic_id if opportunity is None?

cc: @seanlip

Reply:
I agree that `None` would be better than empty string, yup.

But does a default value make sense here, though -- would there ever be a case when there is no topic ID? If not then would throwing an error (or potentially ignoring, if we're sure that any models hitting this check can be ignored) be the right thing to do instead?

Reply:
In the `ExplorationOpportunitySummaryModel` topic_id is not nullable. I think here, the topic_id would be empty, if the corresponding opportunity is not available anymore.

<img width="617" alt="image" src="https://github.com/oppia/oppia/assets/41409951/d33a3f19-1c65-41f8-96d3-43bb6c07495c">

Reply:
But if the opportunity is not available any more then why would we have an ExplorationOpportunitySummaryModel for it? I still think non-nullable is the correct thing conceptually -- unless there's a legitimate case which null represents.